core/src/components/input/input.scss

```diff
@@ -659,8 +659,21 @@
 
 ::slotted([slot="start"]) {
   margin-inline-end: $form-control-label-margin;
+  margin-inline-start: 0;
 }
 
 ::slotted([slot="end"]) {
   margin-inline-start: $form-control-label-margin;
+  margin-inline-end: 0;
 }
+
+::slotted(ion-button[slot="start"].button-has-icon-only),
+::slotted(ion-button[slot="end"].button-has-icon-only) {
+  --border-radius: 50%;
+  --padding-start: 0;
+  --padding-end: 0;
+  --padding-top: 0;
+  --padding-bottom: 0;
+  
+  aspect-ratio: 1;
```

Comment:
Let's add a comment of why this is needed. Since it's connected to the hover and focus, it might get accidentally removed because it's not evident on first glance.
The same for the others.

Reply:
Added in https://github.com/ionic-team/ionic-framework/pull/28676/commits/582d4827e306d8a97bab5803cdcfc94a5bf98401 👍